oai_agents/gym_environments/base_overcooked_env.py

```diff
@@ -243,11 +249,14 @@ def step(self, action):
 
         self.state, reward, done, info = self.env.step(joint_action)
         if self.shape_rewards and not self.is_eval_env:
-            ratio = min(self.step_count * self.args.n_envs / 1e7, 1)
-            sparse_r = sum(info['sparse_r_by_agent'])
-            shaped_r = info['shaped_r_by_agent'][self.p_idx] if self.p_idx else sum(info['shaped_r_by_agent'])
-            reward = sparse_r * ratio + shaped_r * (1 - ratio)
-
+            ratio = min(self.step_count * self.args.n_envs / 1e7, 0.5)
+            group_sparse_r = sum(info['sparse_r_by_agent'])
+            group_shaped_r = sum(info['shaped_r_by_agent'])               
+            sparse_r = info['sparse_r_by_agent'][self.p_idx] if self.p_idx is not None else group_sparse_r
+            shaped_r = info['shaped_r_by_agent'][self.p_idx] if self.p_idx is not None else group_shaped_r
+            p_reward = group_sparse_r * ratio + shaped_r * (1 - ratio)
+            group_reward = (1/self.mdp.num_players) * (self.mdp.num_players * group_sparse_r * ratio + group_shaped_r * (1 - ratio))
```

Comment:
I think all of these calculations should be moved to `Class Learner` you could add a function called `extract_reward`:

```
Class Learner:
    Class LearnerType:
              Helper = 'helper' 
              ....

    def __init__(self, learner_type):
          self.learner_type = learner_type

   def extract_reward(self, p_idx, env_info):
         self.personal_shaped_reward = ..
         self.group_shaped_reward = ..
        # Basically add all the calculations from line 252 to 258
```
 
And then in each of your Learner classes, for example Helper you could do:
```
def calculate_reward(self, p_idx, env_info):
     super().extract_reward(p_idx, env_info)
     return Learner.magnifier * (1/3 * self.individual_reward + 2/3 * self.group_reward)

```
 I think this changes would make the code cleaner